packages/system/src/Stream/Stream/filter.ts

```diff
@@ -0,0 +1,17 @@
+import * as Array_ from "../../Array"
+import type { Predicate } from "../../Function"
+import { pipe } from "../../Function"
+import type { Stream } from "./definitions"
+import { mapChunks } from "./mapChunks"
+
+/**
+ * Applies the predicate to each element and allows passing elements 
+ * to reach the output of this stream.
+ */
+export const filter = <O>(f: Predicate<O>) => <R, E>(
+  self: Stream<R, E, O>
+): Stream<R, E, O> =>
+  pipe(
+    self,
+    mapChunks((o) => Array_.chain_(o, (o) => (f(o) ? [o] : [])))
```

Comment:
It seems that it is possible to use curried version `Array_.chain` instead of uncurried one.

Reply:
we should use here the Array filter function as they do in zio:
https://github.com/zio/zio/blob/master/streams/shared/src/main/scala/zio/stream/ZStream.scala#L1245